pinot-core/src/main/java/org/apache/pinot/core/startree/StarTreeUtils.java

```diff
@@ -176,11 +178,17 @@ public static Map<String, List<CompositePredicateEvaluator>> extractPredicateEva
    * </ul>
    */
   public static boolean isFitForStarTree(StarTreeV2Metadata starTreeV2Metadata,
-      AggregationFunctionColumnPair[] aggregationFunctionColumnPairs, @Nullable ExpressionContext[] groupByExpressions,
-      Set<String> predicateColumns) {
+      List<Pair<AggregationFunction, AggregationFunctionColumnPair>> aggregations,
+      @Nullable ExpressionContext[] groupByExpressions, Set<String> predicateColumns) {
     // Check aggregations
-    for (AggregationFunctionColumnPair aggregationFunctionColumnPair : aggregationFunctionColumnPairs) {
-      if (!starTreeV2Metadata.containsFunctionColumnPair(aggregationFunctionColumnPair)) {
+    for (Pair<AggregationFunction, AggregationFunctionColumnPair> aggregation : aggregations) {
+      AggregationFunction function = aggregation.getLeft();
+      AggregationFunctionColumnPair functionColumnPair = aggregation.getRight();
+      if (!starTreeV2Metadata.containsFunctionColumnPair(functionColumnPair)) {
```

Comment:
Similar to map access, we can directly get the `AggregationSpec` and do a null check to save one lookup